### PR TITLE
Package all files into a single dist folder

### PR DESCRIPTION
**Pull request by barelyhuman**
### Changes 
- Modify Rollup to output to a single folder
- Modify rollup typescript config
- Rename exports to point to `.mjs` in exports
- Added nextjs to the sandbox generators

### CODE REVIEW

**Comment by barelyhuman** [comment]
Existing working sandboxes for the same changes

zustand code sandbox | link
---|---
React | [Link](https://codesandbox.io/s/react-forked-639me?file=/src/App.js:23-353)
React TS | [Link](https://codesandbox.io/s/react-typescript-forked-i2vem)
React Snowpack | [Link](https://codesandbox.io/s/react-snowpack-forked-nypv7)
React Parcel | [Link](https://codesandbox.io/s/react-parcel-forked-ef90w?file=/src/index.js)
Webpack 5 Next | [Link](https://codesandbox.io/s/inspiring-pasteur-tofvl?file=/pages/index.js:0-228)

File: rollup.config.js [hunk 1]
```diff
@@ -27,6 +27,8 @@ function getEsbuild(target) {
     minify: false,
     target,
     tsconfig: path.resolve('./tsconfig.json'),
+    allowSyntheticDefaultImports: true,
+    module: 'esnext',
```
Comment (dai-shi) [inline rollup.config.js:31]: what problem do these solve?

File: rollup.config.js [hunk 2]
```diff
@@ -37,7 +39,13 @@ function createDeclarationConfig(input, output) {
       dir: output,
     },
     external,
-    plugins: [typescript({ declaration: true, outDir: output })],
+    plugins: [
+      typescript({
+        declaration: true,
+        emitDeclarationOnly: true,
```
Comment (dai-shi) [inline rollup.config.js:45]: What happens without this?

**Review by dai-shi — COMMENTED** [review]
Thanks for your work. Really appreciated.
I'd like to ask some questions for clarification.

File: rollup.config.js (hunk 1 above)
Comment (barelyhuman) [reply rollup.config.js]: This one, I missed to reset, was in correspondence to the tsconfig change I made earlier

File: rollup.config.js (hunk 2 above)
Comment (barelyhuman) [reply rollup.config.js:43]: emitDeclarationOnly: true => only creates `src/index.d.ts` and other `.d.ts` files, if you remove that, an additional `index.js` is also created, which overrides the `index.js` generated by rollup, thus creating disparity into the expected output and actual output of the rollup config.

Comment (dai-shi) [reply rollup.config.js:43]: wow, really. so, it must have been a long-existed issue.

Comment (barelyhuman) [reply rollup.config.js:43]: Maybe, I observed it to be happening during the initial ESM PR

Comment (dai-shi) [reply rollup.config.js:43]: I confirmed. The js files created by tsc seem overwritten by files by rollup. So, there has been no issues. It's nice to have, though.

**Review by dai-shi — APPROVED** [review]
thanks for the updates. please update https://github.com/pmndrs/jotai/pull/657 and create a small follow-up PR in valtio repo.

**Pull request merged**